### PR TITLE
Adding attachment info and upload of analysis

### PR DESCRIPTION
**Pull request by shawn-hurley**
Allowing us to get more info when from the task for seeing what is happening in the addon, specfically when this fails

### CODE REVIEW

File: analysis/analysis_test.go [hunk 1]
```diff
@@ -149,12 +150,26 @@ func TestApplicationAnalysis(t *testing.T) {
...
 			if task.State != "Succeeded" {
 				t.Error("Analyze Task failed. Details:")
-				pp.Println(task)
 				pp.Println(tasks)
+				dir, err := os.MkdirTemp("", "attachments")
+				if err != nil {
+					t.Error(err)
+				}
+				printAllAtachmentsOnTask(task, dir)
+				// If the task was unsussfull there is no reason to continue execution.
```
Comment (jmontleon) [inline analysis/analysis_test.go:171]: Was that meant to be unsuccessful?

Comment (shawn-hurley) [reply analysis/analysis_test.go]: I can not help but add typos sorry

**Review by jmontleon — APPROVED** [review]
ACK other than there is still a typo: `unsuccessfulthere`

**Review by pranavgaikwad — APPROVED** [review]

**Review by aufi — APPROVED** [review]
LGTM

**Pull request merged**